test/unistd/access.c

```diff
@@ -65,6 +66,57 @@ int main() {
   printf("F_OK(%s): %d\n", "renamedfile", faccessat(AT_FDCWD, "renamedfile", F_OK, 0));
   printf("errno: %d\n", errno);
 
+  EM_ASM(
+    var fchmodstream = FS.open("fchmodtest", "r");
+#if WASMFS
+    FS.fchmod(fchmodstream, 0o000);
+#else
+    FS.fchmod(fchmodstream.fd, 0o000);
+#endif
+  );
+  struct stat fileStats;
+  stat("fchmodtest", &fileStats);
+  assert(!(fileStats.st_mode & 0777));
```

Comment:
Can this not test if the mode is 0? I have a vague memory that there is some other stuff in the mode... If so, can we test that the mode was 0777 before? That would make it clear that the test is proving that the `*chmod` lines had an actual effect.

Reply:
Oh, that makes sense. I added a test to check the permissions of `fchmodtest` before calling `fchmod`. I think `NODEFS` and `NODERAWFS` give different default permissions with writeFile, so I added an ifdef to account for that.